aslam_cv/aslam_cameras_april/test/GridCalibrationTargetAprilgridv2Test.cpp

```diff
@@ -0,0 +1,102 @@
+#include <iostream>
+#include <vector>
+#include <chrono>
+#include <opencv2/opencv.hpp>
+#include <aslam/cameras/GridCalibrationTargetAprilgrid.hpp>
+#include <aslam/cameras/GridCalibrationTargetAprilgridv2.hpp>
+#include <gtest/gtest.h>
+
+using namespace std;
+
+// Create a synthetic AprilTag grid image (grayscale) for tests.
+// Parameters use pixels for sizes and spacing. Returns a CV_8UC1 image.
+cv::Mat createTestAprilGrid(int gridRows,
+                            int gridCols,
+                            int markerSize = 200,
+                            int margin = 40,
+                            int spacing = 40,
+                            int borderBits = 2) {
+
+  cv::Ptr<cv::aruco::Dictionary> dictionary =
+      cv::aruco::getPredefinedDictionary(cv::aruco::DICT_APRILTAG_36h11);
+
+  const int canvasWidth = margin * 2 + gridCols * (markerSize) + (gridCols - 1) * spacing;
+  const int canvasHeight = margin * 2 + gridRows * (markerSize) + (gridRows - 1) * spacing;
+
+  // white background
+  cv::Mat canvas(canvasHeight, canvasWidth, CV_8UC1, cv::Scalar(255));
+
+  // draw markers with ids sequential left-to-right, top-to-bottom
+  for (int r = 0; r < gridRows; ++r) {
+    for (int c = 0; c < gridCols; ++c) {
+      int id = r * gridCols + c;
+      cv::Mat marker;
+      cv::aruco::drawMarker(dictionary, id, markerSize, marker, borderBits);
+
+      int x = margin + c * (markerSize + spacing);
+      int y = margin + r * (markerSize + spacing);
+      marker.copyTo(canvas(cv::Rect(x, y, markerSize, markerSize)));
+    }
+  }
+
+  return canvas;
+}
+
+// Validate that both versions of the AprilTag detector produce similar results
+TEST (DetectorComparisonTest, Compare) {
+  // 1. Generate test image with a 2x2 AprilTag grid and save it to disk
+  constexpr int gridRows = 6;
```

Comment:
@frneer nit^N: consider hungarian notation for constants e.g.:
```suggestion
  constexpr int kGridRows = 6;
```